components/dashboard/src/settings/Preferences.tsx

```diff
@@ -118,50 +104,48 @@ export default function Preferences() {
-                            checked={useDesktopIde}
-                            onChange={(evt) => actuallySetUseDesktopIde(evt.target.checked)} />
+                {browserIdeOptions && <>
+                    <h3>Browser Editor</h3>
+                    <p className="text-base text-gray-500 dark:text-gray-400">Choose the default editor for opening workspaces in the browser.</p>
+                    <div className="my-4 gap-4 flex flex-wrap">
+                        {
+                            browserIdeOptions.map(([id, option]) => {
+                                const selected = defaultIde === id;
+                                const onSelect = () => actuallySetDefaultIde(id);
+                                return renderIdeOption(option, selected, onSelect);
+                            })
+                        }
                     </div>
-                    {useDesktopIde && <>
-                        <div className="my-4 space-x-4 flex">
-                            {
-                                desktopIdeOptions.map(([id, option]) => {
-                                    const selected = defaultDesktopIde === id;
-                                    const onSelect = () => actuallySetDefaultDesktopIde(id);
-                                    return renderIdeOption(option, selected, onSelect);
-                                })
-                            }
-                        </div>
-
-                        {ideOptions.options[defaultDesktopIde].notes &&
-                            <InfoBox className="my-5 max-w-2xl"><ul>
-                                {ideOptions.options[defaultDesktopIde].notes?.map((x, idx) => <li className={idx > 0 ? "mt-2" : ""}>{x}</li>)}
-                            </ul></InfoBox>
+                    {ideOptions.options[defaultIde] && ideOptions.options[defaultIde].notes &&
+                        <InfoBox className="my-5 max-w-2xl"><ul>
+                            {ideOptions.options[defaultIde].notes?.map((x, idx) => <li className={idx > 0 ? "mt-2" : ""}>{x}</li>)}
+                        </ul></InfoBox>
+                    }
+                </>}
+                {desktopIdeOptions && <>
+                    <h3 className="mt-12">Desktop Editor</h3>
+                    <p className="text-base text-gray-500 dark:text-gray-400">Optionally, choose the default desktop editor for opening workspaces.</p>
+                    <div className="my-4 gap-4 flex flex-wrap">
+                        {
+                            desktopIdeOptions.map(([id, option]) => {
+                                let selected = defaultDesktopIde === id;
+                                const onSelect = () => actuallySetDefaultDesktopIde(id);
+                                if (id === DesktopNoneId) {
+                                    selected = defaultDesktopIde === DesktopNoneId
```

Comment:
This isn't needed, right? (I think `selected` above will already be correct when `id === DesktopNoneId`)

```suggestion
```